#EE282 HW#3

**To download the all chromosomes and md5sum fasta files into a new folder:**

```
mkdir melanogaster_data
cd melanogaster_data
wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/fasta/dmel-all-chromosome-r6.24.fasta.gz
wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/fasta/md5sum.txt

```
**File integrity:**

```
md5sum *.gz > md5check_genome.txt
diff md5{sum,check_genome}.txt
mv md5sum.txt md5sum_genome.txt
```
Make sure the chromosomes file does not appear in the output, but the files you did not download will because they are in the md5sum file.

**To calculate number of nucleotides, number of N's, and number of sequences:**

```
faSize dmel-all-chromosome-r6.24.fasta.gz
```

**To download the gtf annotation file and md5sum:**

```
wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/gtf/dmel-all-r6.24.gtf.gz
wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/gtf/md5sum.txt
```
**Annotation file integrity:**

```
md5sum *.gz > md5check_annotation.txt
diff md5{sum,check_annotation}.txt
mv md5sum.txt md5sum_annotation.txt
```
**Print a summary report with the following information:**
	
1. Total number of features of each type, sorted from the most common to the least common


```
zcat dmel-all-r6.24.gtf.gz | awk '{print $3}'| sort | uniq -c | sort -rn > summary_features.txt
```

2. Total number of genes per chromosome arm (X, Y, 2L, 
2R, 3L, 3R, 4)

```
zcat dmel-all-r6.24.gtf.gz | awk '{print $1 "\t" $3}' | grep 'gene' | awk '{print $1}' | sort | uniq -c | sort -rn > summary_genes.txt
```
